File: src/core/session_state.nim
import tables
import models/command

type
  SessionState* = ref object
    availableCommands*: seq[string]
    environment*: Table[string, string]
    aliases*: Table[string, Command]
    homeDir*: string
    curDir*: string
    history*: seq[Command]
    cont*: bool

proc newSessionState*(environment: Table[string, string], homeDirectory: string,
    currentDirectory: string): SessionState =
  SessionState(
    availableCommands: @[],
    environment: environment,
    aliases: initTable[string, Command](),
    homeDir: homeDirectory,
    curDir: currentDirectory,
    history: @[],
    cont: true,
  )

File: src/core/session_state.rs
use std::collections::HashMap;
use crate::models::command::Command;

pub struct SessionState {
    pub available_commands: Vec<String>,
    pub environment: HashMap<String, String>,
    pub aliases: HashMap<String, Command>,
    pub home_dir: String,
    pub cur_dir: String,
    pub history: Vec<Command>,
    pub cont: bool,
}

impl SessionState {
    pub fn new(
        environment: HashMap<String, String>,
        home_directory: String,
        current_directory: String,
    ) -> Self {
        SessionState {
            available_commands: Vec::new(),
            environment,
            aliases: HashMap::new(),
            home_dir: home_directory,
            cur_dir: current_directory,
            history: Vec::new(),
            cont: true,
        }
    }
}
